lib/ember-test-helpers/test-module.js

```diff
@@ -107,7 +107,8 @@ export default Klass.extend({
     function nextStep() {
       var step = steps.shift();
       if (step) {
-        return Ember.RSVP.resolve(step.call(context)).then(nextStep);
+        // guard against exceptions, for example missing components referenced from needs.
+        return new Ember.RSVP.Promise(function(ok) {ok(step.call(context));}).then(nextStep);
```

Comment:
@chadhietala your proposal does not work. You can rewrite both the original code and your proposal as the following:

```javascript
var current = step.call(context); // throws exception, promise code never reached
return Ember.RSVP.resolve(current).then(..)
```

The call to `step.call()` will happen before the promise, even though it is written on the same line. With the proposed change the promise is active before the throwing call and can setup a try-catch to catch the exception. 

Note that `promise.resolve()` creates a new function anyways so there is no change in performance or memory allocation.